RRS-108 Create Landlord Profile Page

OttoChu opened the pull request:
I have done the basic layout for the landlord profile page and can fetch the correct data from the database. 
The only thing missing would be the links to the relevant property when showing the list of properties owned by the landlord.

We just need someone on the front-end team to make it look good.

OttoChu committed:
Created example landlord profile page

OttoChu committed:
Added check user logged in

OttoChu committed:
Merge branch 'development' into RRS-108-Create-landlord-profile-page

OttoChu committed:
Able to get real data from database

LukeHosk commented:
I was having a look at testing this and got distracted by the fact that the users i was creating didn't have profiles despite merging my branch. fixed that, will have another look at this now

LukeHosk commented on app/profiles/landlord/page.tsx:
```diff
@@ -0,0 +1,91 @@
+import { cookies } from "next/headers"
+import { NextPage } from "next"
+
+const landlordProfilePage: NextPage = async ({
+    searchParams
+}: {
+    searchParams?: {
+        landlordId?: string,
+    }
+}) => {
```
I think think this is a case where we would be better off using a [dynamic route](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes) instead of search params. But it doesn't make a huge difference.

LukeHosk commented on app/profiles/landlord/page.tsx:
```diff
@@ -0,0 +1,91 @@
+
+    // check if a landlord with the provided id exists and get their info
+    const { data: landlordData, error: landlordError } = await supabase
+        .from('user_profiles')
```
I think this should be `.from('landlord_public_profiles')`

`user_profiles` are currently only visible to that user (which is what we want)

OttoChu replied:
But then I can't get the details for the landlord because "first_name", "last_name" and "email is stored in userprofile table

LukeHosk replied:
If we want to be able to show that to all users, then we should be storing it in the landlord profiles, as the user profiles are private as per the schema plan:
<img width="1196" alt="Screenshot 2024-02-22 at 00 54 08" src="https://github.com/CM20254-Group-7/Rental_Review/assets/74842299/ad227b9f-dab9-458d-b96a-eab6ccd0c623">

I also believe we have dropped the first and last name from the user profiles, as we are not collecting it from users who are not landlords.

it might also be beneficial to have a separate data point for the landlords public facing email if they want to show a different one to the one they use to login

LukeHosk replied:
I also suggested there that we might want to make the landlord's real name private and give them a username for the public but i don't think we want to do it that way

LukeHosk commented on app/profiles/landlord/page.tsx:
```diff
@@ -0,0 +1,91 @@
+        <div>
+            <h1>ERROR: Unable to fetch landlord properties</h1>
+        </div>
+    ); else if (!landlordProperties) return (
```
do we want to hide the profile if the landlord has no properties?

I would have thought we wanted to just show the profile details

LukeHosk commented on app/profiles/landlord/page.tsx:
```diff
@@ -0,0 +1,91 @@
+            <p>Name: {landlordData.first_name + ' ' + landlordData.last_name}</p>
+            <p>Email: {landlordData.email}</p>
+            <p>Created at: {landlordData.created_at}</p>
+            {/* CHANGE THIS */}
```
what needs changing here?

are we including fields like the bio?

LukeHosk commented on app/profiles/landlord/page.tsx:
```diff
@@ -0,0 +1,91 @@
+        .eq('user_id', landlordId)
+        .single()
+
+    if (landlordError || !landlordData) return (
+        <div>
+            <h1>ERROR: Landlord profile not found</h1>
+        </div>
+    )
```
you might want to consider using [not-found](https://nextjs.org/docs/app/api-reference/functions/not-found) here

LukeHosk reviewed (changes requested):
This currently only works when logged in as the landlord who's page you are viewing

OttoChu committed:
Fixed few bugs

- Still show profile if the landlord owns no properties
- used noFound when landlord doesn't exist
- Added bio and made what to change more understandable

OttoChu committed:
Added the dynamic routing feature

LukeHosk commented on app/profiles/[landlordId]/page.tsx:
```diff
@@ -0,0 +1,103 @@
+export default async function landlordProfilePage( {params }: { params: { landlordId: string } }) {
+    // check if a landlord id was provided
+    const landlordId = params.landlordId
+    if (!landlordId) return (
+        <div>
+            <h1>ERROR: No Landlord Id provided</h1>
+        </div>
+    )
```
I believe this check is un-necessary, as the case where there is no landlord id provided will go to the other page you created (or 404 if that one wan't defined)

OttoChu committed:
Changed where landlord details are stored

LukeHosk committed:
Merge remote-tracking branch 'origin/development' into RRS-108-Create-landlord-profile-page

LukeHosk committed:
fields already removed from user table

LukeHosk committed:
type file index issue fix

LukeHosk commented:
Fixed a small conflict where multiple migrations were trying to drop the name fields from the user table

LukeHosk reviewed (approved):
Tested myself and seems to work as intended.

Would be nice if you could add unit tests, but we can merge for now.
